src/silvimetric/resources/metric.py

```diff
@@ -87,12 +88,24 @@ def entry_name(self, attr: str) -> str:
         """Name for use in TileDB and extract file generation."""
         return f'm_{attr}_{self.name}'
 
-    def sanitize_and_run(self, d, *args):
-        # args come in as a value wrapped in one index of a 2D dataframe
-        # we need to remove the wrapping and pass the values to the method
-        # to make things easier
-        a = tuple( a.values[0][0] for a in args)
-        return self._method(d, *a)
+    def sanitize_and_run(self, d, locs, args):
+        # Args are the return values of previous DataFrame aggregations.
+        # In order to access the correct location, we need a map of groupby
+        # indices to their locations and then grab the correct index from args
+
+        attr = d.name
+        attrs = [a.entry_name(attr) for a in self.dependencies]
+
+        if isinstance(args, pd.DataFrame):
+            idx = locs.loc[d.index[0]]
+            xi = idx.xi
+            yi = idx.yi
+            with mutex:
```

Comment:
doesn't this serialize the entire frame to the `local()`'s context? Is this what you actually want here?